eZ/Publish/Core/Repository/Tests/Service/Mock/Base.php

```diff
@@ -333,4 +338,24 @@ protected function getLanguageResolverMock(): LanguageResolver
 
         return $this->languageResolverMock;
     }
+
+    /**
+     * @param string[] $methods
+     *
+     * @return \eZ\Publish\Core\Repository\Mapper\RoleDomainMapper|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected function getRoleDomainMapperMock(array $methods = []): RoleDomainMapper
+    {
+        if ($this->roleDomainMapperMock === null) {
+            $mockBuilder = $this->getMockBuilder(RoleDomainMapper::class);
+            if (!empty($methods)) {
+                $mockBuilder->onlyMethods($methods);
+            }
+            $this->roleDomainMapperMock = $mockBuilder
+                ->disableOriginalConstructor()
+                ->getMock();
+        }
+
+        return $this->roleDomainMapperMock;
+    }
```

Comment:
Moved from child class as now it's needed here as well. Not 1:1 because `setMethods` is deprecated. Diff could be a bit confusing, because passing `$methods` directly will lead to "no methods" with an empty array, rather than what we've expected here.